### PR TITLE
feat(boot): add more helper methods for model and application booters

### PR DESCRIPTION
**Pull request by raymondfeng**
1. This further simplifies how a main application can boot sub-applications. The idea was from @bajtos in one of the comments (I cannot find it anymore :-).

2. Add `model()` to `RepositoryMixin` and use it for model booter (https://github.com/strongloop/loopback-next/pull/5378#discussion_r424980840)

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/boot/src/__tests__/acceptance/component-application.booter.acceptance.ts [hunk 1]
```diff
@@ -97,6 +124,14 @@ describe('component application booter acceptance tests', () => {
     }
   }
 
+  class MainAppWithSubAppBooter extends BootMixin(Application) {
+    constructor() {
+      super();
+      this.projectRoot = __dirname;
+      this.applicationBooter(app);
```
Comment (bajtos) [inline packages/boot/src/__tests__/acceptance/component-application.booter.acceptance.ts:131]: Should the method name start with a verb, e.g. `this.addApplicationBooter`?

File: packages/boot/src/mixins/boot.mixin.ts [hunk 2]
```diff
@@ -104,6 +106,18 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
       );
     }
 
+    /**
+     * Register a booter to boot a sub-application
+     * @param subApp - A sub-application with artifacts to be booted
+     * @param filter - A binding filter to select what bindings from the sub
+     * application should be added to the main application.
```
Comment (bajtos) [inline packages/boot/src/mixins/boot.mixin.ts:113]: Can you please briefly mention what is the default behavior when no filter is provided? Are we going to load all bindings or no bindings?

File: packages/repository/src/mixins/repository.mixin.ts [hunk 3]
```diff
@@ -141,6 +141,16 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
...
     }
 
+    /**
+     * Register a model class
+     * @param modelClass - Model class
+     */
+    model<M extends Class<unknown>>(modelClass: M) {
+      return this.bind<M>(`models.${modelClass.name}`)
+        .to(modelClass)
+        .tag('model');
```
Comment (bajtos) [inline packages/repository/src/mixins/repository.mixin.ts:151]: This is definitely an improvement :+1:

However, it still does not allow components (extensions) to create a model binding that can be added to `this.bindings` exported by the component. Have you considered adding also `createModelClassBinding` as I proposed in my earlier comment?

File: packages/boot/src/__tests__/acceptance/component-application.booter.acceptance.ts (hunk 1 above)
Comment (raymondfeng) [reply packages/boot/src/__tests__/acceptance/component-application.booter.acceptance.ts:131]: Again, I choose a noun to be consistent with other methods.

File: packages/repository/src/mixins/repository.mixin.ts (hunk 3 above)
Comment (raymondfeng) [reply packages/repository/src/mixins/repository.mixin.ts]: Added.

**Pull request merged**